File: src/main/java/no/nav/syfo/mocks/AdresseregisterV1Mock.java
package no.nav.syfo.mocks;


import com.microsoft.schemas._2003._10.serialization.arrays.WSArrayOfint;
import no.nhn.register.certificatedetails.WSArrayOfCertificateDetails;
import no.nhn.register.certificatedetails.WSCertificateDetails;
import no.nhn.register.certificatesearchresult.WSArrayOfLdapSearchProvider;
import no.nhn.register.certificatesearchresult.WSCertificateSearchResult;
import no.nhn.register.common.WSArrayOfEntityLogEntry;
import no.nhn.register.common.WSCode;
import no.nhn.register.common.WSPeriod;
import no.nhn.register.communicationparty.*;
import org.datacontract.schemas._2004._07.system_collections.WSArrayOfKeyValuePairOfintArrayOfCodeUO9UWExd;

import javax.xml.datatype.XMLGregorianCalendar;

public class AdresseregisterV1Mock implements ICommunicationPartyService {


    @Override
    public void updateDepartmentDetails(WSDepartmentUpdate org) throws ICommunicationPartyServiceUpdateDepartmentDetailsValidationFaultFaultFaultMessage, ICommunicationPartyServiceUpdateDepartmentDetailsGenericFaultFaultFaultMessage {

    }

    @Override
    public WSService getServiceDetails(Integer herId) throws ICommunicationPartyServiceGetServiceDetailsGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSSearchResult search(WSCommunicationPartySearch search) throws ICommunicationPartyServiceSearchGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSOrganization getOrganizationDetails(Integer herId) throws ICommunicationPartyServiceGetOrganizationDetailsGenericFaultFaultFaultMessage {
        return new WSOrganization()
                .withDepartments(new WSArrayOfDepartment()
                        .withDepartments(new WSDepartment()
                                .withParentHerId(12345678))
                );
    }

    @Override
    public byte[] getCertificateForValidatingSignatureLdap(String ldapUrl) throws ICommunicationPartyServiceGetCertificateForValidatingSignatureLdapValidationFaultFaultFaultMessage, ICommunicationPartyServiceGetCertificateForValidatingSignatureLdapGenericFaultFaultFaultMessage {
        return new byte[0];
    }

    @Override
    public void updateOrganizationIKSAssociations(Integer organizaitonHerId, WSArrayOfint herIdsForAssociatedIKS) throws ICommunicationPartyServiceUpdateOrganizationIKSAssociationsValidationFaultFaultFaultMessage, ICommunicationPartyServiceUpdateOrganizationIKSAssociationsGenericFaultFaultFaultMessage {

    }

    @Override
    public void enableTransportOnOrganization(Integer herId, String transportType) throws ICommunicationPartyServiceEnableTransportOnOrganizationGenericFaultFaultFaultMessage {

    }

    @Override
    public WSArrayOfOrganization getAssosiatedIKSParentOrganizations(Integer herId) throws ICommunicationPartyServiceGetAssosiatedIKSParentOrganizationsGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSArrayOfKeyValuePairOfintArrayOfCodeUO9UWExd getCustomAttributes(WSArrayOfint herIds) {
        return null;
    }

    @Override
    public WSCommunicationPartyStatistics getCommunicationPartyStatistics() throws ICommunicationPartyServiceGetCommunicationPartyStatisticsGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSCommunicationParty getCommunicationPartyDetails(Integer herId) throws ICommunicationPartyServiceGetCommunicationPartyDetailsGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSArrayOfOrganization getOrganizationsHavingCode(WSCode code) throws ICommunicationPartyServiceGetOrganizationsHavingCodeGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSCertificateDetails getCertificateDetailsForEncryption(Integer herId) throws ICommunicationPartyServiceGetCertificateDetailsForEncryptionValidationFaultFaultFaultMessage, ICommunicationPartyServiceGetCertificateDetailsForEncryptionGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSCode addCustomAttribute(Integer herId, WSCode code) throws ICommunicationPartyServiceAddCustomAttributeGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSCertificateSearchResult searchCertificatesByLdapUrl(String ldapUrl) throws ICommunicationPartyServiceSearchCertificatesByLdapUrlValidationFaultFaultFaultMessage, ICommunicationPartyServiceSearchCertificatesByLdapUrlGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public byte[] getCertificateForEncryptionByOrganizationNumber(Integer organizationNumber) throws ICommunicationPartyServiceGetCertificateForEncryptionByOrganizationNumberValidationFaultFaultFaultMessage, ICommunicationPartyServiceGetCertificateForEncryptionByOrganizationNumberGenericFaultFaultFaultMessage {
        return new byte[0];
    }

    @Override
    public WSArrayOfCommunicationParty searchById(String id) throws ICommunicationPartyServiceSearchByIdGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public byte[] getCertificateForEncryptionLdap(String ldapUrl) throws ICommunicationPartyServiceGetCertificateForEncryptionLdapValidationFaultFaultFaultMessage, ICommunicationPartyServiceGetCertificateForEncryptionLdapGenericFaultFaultFaultMessage {
        return new byte[0];
    }

    @Override
    public byte[] getCertificateForValidatingSignature(Integer herId) throws ICommunicationPartyServiceGetCertificateForValidatingSignatureValidationFaultFaultFaultMessage, ICommunicationPartyServiceGetCertificateForValidatingSignatureGenericFaultFaultFaultMessage {
        return new byte[0];
    }

    @Override
    public void updateOrganizationPersonDetails(WSOrganizationPersonUpdate p) throws ICommunicationPartyServiceUpdateOrganizationPersonDetailsValidationFaultFaultFaultMessage, ICommunicationPartyServiceUpdateOrganizationPersonDetailsGenericFaultFaultFaultMessage {

    }

    @Override
    public void updateOrganizationDetails(WSOrganizationUpdate org) throws ICommunicationPartyServiceUpdateOrganizationDetailsValidationFaultFaultFaultMessage, ICommunicationPartyServiceUpdateOrganizationDetailsGenericFaultFaultFaultMessage {

    }

    @Override
    public WSService createService(WSServiceCreate t) throws ICommunicationPartyServiceCreateServiceValidationFaultFaultFaultMessage, ICommunicationPartyServiceCreateServiceGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSOrganizationPerson getOrganizationPersonDetails(Integer herId) throws ICommunicationPartyServiceGetOrganizationPersonDetailsGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSCertificateDetails getCertificateDetailsForValidatingSignature(Integer herId) throws ICommunicationPartyServiceGetCertificateDetailsForValidatingSignatureValidationFaultFaultFaultMessage, ICommunicationPartyServiceGetCertificateDetailsForValidatingSignatureGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSArrayOfOrganization getOrganizationsOwnedBy(Integer herId) throws ICommunicationPartyServiceGetOrganizationsOwnedByGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSCommunicationParty getGPCommunicationParty(String ssn) throws ICommunicationPartyServiceGetGPCommunicationPartyGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public String ping() {
        return null;
    }

    @Override
    public WSOrganizationUpdateStatistics getOrganizationUpdateStatistics(XMLGregorianCalendar fromDate, XMLGregorianCalendar toDate) throws ICommunicationPartyServiceGetOrganizationUpdateStatisticsGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSArrayOfCertificateDetails searchCertificatesForValidatingSignature(String sn, String cn, String ou) throws ICommunicationPartyServiceSearchCertificatesForValidatingSignatureValidationFaultFaultFaultMessage, ICommunicationPartyServiceSearchCertificatesForValidatingSignatureGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public void updateServiceDetails(WSServiceUpdate t) throws ICommunicationPartyServiceUpdateServiceDetailsValidationFaultFaultFaultMessage, ICommunicationPartyServiceUpdateServiceDetailsGenericFaultFaultFaultMessage {

    }

    @Override
    public byte[] getCertificate(String ldapUrl) throws ICommunicationPartyServiceGetCertificateValidationFaultFaultFaultMessage, ICommunicationPartyServiceGetCertificateGenericFaultFaultFaultMessage {
        return new byte[0];
    }

    @Override
    public byte[] getCertificateForEncryption(Integer herId) throws ICommunicationPartyServiceGetCertificateForEncryptionValidationFaultFaultFaultMessage, ICommunicationPartyServiceGetCertificateForEncryptionGenericFaultFaultFaultMessage {
        return new byte[0];
    }

    @Override
    public WSArrayOfLdapSearchProvider getLdapSearchProviders() throws ICommunicationPartyServiceGetLdapSearchProvidersValidationFaultFaultFaultMessage, ICommunicationPartyServiceGetLdapSearchProvidersGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public void updateCoverageAreas(Integer herId, WSArrayOfCoverageInfo coverageAreas) {

    }

    @Override
    public byte[] getCertificateForValidatingSignatureByOrganizationNumber(Integer organizationNumber) throws ICommunicationPartyServiceGetCertificateForValidatingSignatureByOrganizationNumberValidationFaultFaultFaultMessage, ICommunicationPartyServiceGetCertificateForValidatingSignatureByOrganizationNumberGenericFaultFaultFaultMessage {
        return new byte[0];
    }

    @Override
    public WSDepartment getDepartmentDetails(Integer herId) throws ICommunicationPartyServiceGetDepartmentDetailsGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSArrayOfCertificateDetails searchCertificatesForEncryption(String sn, String cn, String ou) throws ICommunicationPartyServiceSearchCertificatesForEncryptionValidationFaultFaultFaultMessage, ICommunicationPartyServiceSearchCertificatesForEncryptionGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSArrayOfOrganization getAssosiatedIKSChildOrganizations(Integer herId) throws ICommunicationPartyServiceGetAssosiatedIKSChildOrganizationsGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSOrganization createOrganization(WSOrganizationCreate org) throws ICommunicationPartyServiceCreateOrganizationValidationFaultFaultFaultMessage, ICommunicationPartyServiceCreateOrganizationGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSArrayOfEntityLogEntry getChangeLog(Integer herId) throws ICommunicationPartyServiceGetChangeLogGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSOrganizationPerson createOrganizationPerson(WSOrganizationPersonCreate p) throws ICommunicationPartyServiceCreateOrganizationPersonValidationFaultFaultFaultMessage, ICommunicationPartyServiceCreateOrganizationPersonGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public WSDepartment createDepartment(WSDepartmentCreate org) throws ICommunicationPartyServiceCreateDepartmentValidationFaultFaultFaultMessage, ICommunicationPartyServiceCreateDepartmentGenericFaultFaultFaultMessage {
        return null;
    }

    @Override
    public void setCommunicationPartyValid(Integer herId, WSPeriod period) throws ICommunicationPartyServiceSetCommunicationPartyValidValidationFaultFaultFaultMessage, ICommunicationPartyServiceSetCommunicationPartyValidGenericFaultFaultFaultMessage {

    }

    @Override
    public String getAllHFCommpartiesCsv() {
        return null;
    }

    @Override
    public void removeCustomAttribute(Integer herId, WSCode code) throws ICommunicationPartyServiceRemoveCustomAttributeGenericFaultFaultFaultMessage {

    }
}

File: src/main/kotlin/no/nav/syfo/consumer/ws/adresseregister/AdresseregisterV1Mock.kt
package no.nav.syfo.consumer.ws.adresseregister

import com.microsoft.schemas._2003._10.serialization.arrays.WSArrayOfint
import no.nhn.register.certificatedetails.WSArrayOfCertificateDetails
import no.nhn.register.certificatedetails.WSCertificateDetails
import no.nhn.register.certificatesearchresult.WSArrayOfLdapSearchProvider
import no.nhn.register.certificatesearchresult.WSCertificateSearchResult
import no.nhn.register.common.*
import no.nhn.register.communicationparty.*
import org.datacontract.schemas._2004._07.system_collections.WSArrayOfKeyValuePairOfintArrayOfCodeUO9UWExd
import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty
import org.springframework.stereotype.Service
import javax.xml.datatype.XMLGregorianCalendar

@Service
@ConditionalOnProperty(value = ["mockAdresseregisteretV1"], havingValue = "true")
class AdresseregisterV1Mock : ICommunicationPartyService {
    @Throws(ICommunicationPartyServiceUpdateDepartmentDetailsValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceUpdateDepartmentDetailsGenericFaultFaultFaultMessage::class)
    override fun updateDepartmentDetails(org: WSDepartmentUpdate) {
    }

    @Throws(ICommunicationPartyServiceGetServiceDetailsGenericFaultFaultFaultMessage::class)
    override fun getServiceDetails(herId: Int): WSService? {
        return null
    }

    @Throws(ICommunicationPartyServiceSearchGenericFaultFaultFaultMessage::class)
    override fun search(search: WSCommunicationPartySearch): WSSearchResult? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetOrganizationDetailsGenericFaultFaultFaultMessage::class)
    override fun getOrganizationDetails(herId: Int): WSOrganization {
        return WSOrganization()
            .withDepartments(WSArrayOfDepartment()
                .withDepartments(WSDepartment()
                    .withParentHerId(12345678))
            )
    }

    @Throws(ICommunicationPartyServiceGetCertificateForValidatingSignatureLdapValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceGetCertificateForValidatingSignatureLdapGenericFaultFaultFaultMessage::class)
    override fun getCertificateForValidatingSignatureLdap(ldapUrl: String): ByteArray {
        return ByteArray(0)
    }

    @Throws(ICommunicationPartyServiceUpdateOrganizationIKSAssociationsValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceUpdateOrganizationIKSAssociationsGenericFaultFaultFaultMessage::class)
    override fun updateOrganizationIKSAssociations(organizaitonHerId: Int, herIdsForAssociatedIKS: WSArrayOfint) {
    }

    @Throws(ICommunicationPartyServiceEnableTransportOnOrganizationGenericFaultFaultFaultMessage::class)
    override fun enableTransportOnOrganization(herId: Int, transportType: String) {
    }

    @Throws(ICommunicationPartyServiceGetAssosiatedIKSParentOrganizationsGenericFaultFaultFaultMessage::class)
    override fun getAssosiatedIKSParentOrganizations(herId: Int): WSArrayOfOrganization? {
        return null
    }

    override fun getCustomAttributes(herIds: WSArrayOfint): WSArrayOfKeyValuePairOfintArrayOfCodeUO9UWExd? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetCommunicationPartyStatisticsGenericFaultFaultFaultMessage::class)
    override fun getCommunicationPartyStatistics(): WSCommunicationPartyStatistics? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetCommunicationPartyDetailsGenericFaultFaultFaultMessage::class)
    override fun getCommunicationPartyDetails(herId: Int): WSCommunicationParty? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetOrganizationsHavingCodeGenericFaultFaultFaultMessage::class)
    override fun getOrganizationsHavingCode(code: WSCode): WSArrayOfOrganization? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetCertificateDetailsForEncryptionValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceGetCertificateDetailsForEncryptionGenericFaultFaultFaultMessage::class)
    override fun getCertificateDetailsForEncryption(herId: Int): WSCertificateDetails? {
        return null
    }

    @Throws(ICommunicationPartyServiceAddCustomAttributeGenericFaultFaultFaultMessage::class)
    override fun addCustomAttribute(herId: Int, code: WSCode): WSCode? {
        return null
    }

    @Throws(ICommunicationPartyServiceSearchCertificatesByLdapUrlValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceSearchCertificatesByLdapUrlGenericFaultFaultFaultMessage::class)
    override fun searchCertificatesByLdapUrl(ldapUrl: String): WSCertificateSearchResult? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetCertificateForEncryptionByOrganizationNumberValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceGetCertificateForEncryptionByOrganizationNumberGenericFaultFaultFaultMessage::class)
    override fun getCertificateForEncryptionByOrganizationNumber(organizationNumber: Int): ByteArray {
        return ByteArray(0)
    }

    @Throws(ICommunicationPartyServiceSearchByIdGenericFaultFaultFaultMessage::class)
    override fun searchById(id: String): WSArrayOfCommunicationParty? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetCertificateForEncryptionLdapValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceGetCertificateForEncryptionLdapGenericFaultFaultFaultMessage::class)
    override fun getCertificateForEncryptionLdap(ldapUrl: String): ByteArray {
        return ByteArray(0)
    }

    @Throws(ICommunicationPartyServiceGetCertificateForValidatingSignatureValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceGetCertificateForValidatingSignatureGenericFaultFaultFaultMessage::class)
    override fun getCertificateForValidatingSignature(herId: Int): ByteArray {
        return ByteArray(0)
    }

    @Throws(ICommunicationPartyServiceUpdateOrganizationPersonDetailsValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceUpdateOrganizationPersonDetailsGenericFaultFaultFaultMessage::class)
    override fun updateOrganizationPersonDetails(p: WSOrganizationPersonUpdate) {
    }

    @Throws(ICommunicationPartyServiceUpdateOrganizationDetailsValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceUpdateOrganizationDetailsGenericFaultFaultFaultMessage::class)
    override fun updateOrganizationDetails(org: WSOrganizationUpdate) {
    }

    @Throws(ICommunicationPartyServiceCreateServiceValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceCreateServiceGenericFaultFaultFaultMessage::class)
    override fun createService(t: WSServiceCreate): WSService? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetOrganizationPersonDetailsGenericFaultFaultFaultMessage::class)
    override fun getOrganizationPersonDetails(herId: Int): WSOrganizationPerson? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetCertificateDetailsForValidatingSignatureValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceGetCertificateDetailsForValidatingSignatureGenericFaultFaultFaultMessage::class)
    override fun getCertificateDetailsForValidatingSignature(herId: Int): WSCertificateDetails? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetOrganizationsOwnedByGenericFaultFaultFaultMessage::class)
    override fun getOrganizationsOwnedBy(herId: Int): WSArrayOfOrganization? {
        return null
    }

    override fun ping(): String? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetOrganizationUpdateStatisticsGenericFaultFaultFaultMessage::class)
    override fun getOrganizationUpdateStatistics(fromDate: XMLGregorianCalendar, toDate: XMLGregorianCalendar): WSOrganizationUpdateStatistics? {
        return null
    }

    @Throws(ICommunicationPartyServiceSearchCertificatesForValidatingSignatureValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceSearchCertificatesForValidatingSignatureGenericFaultFaultFaultMessage::class)
    override fun searchCertificatesForValidatingSignature(sn: String, cn: String, ou: String): WSArrayOfCertificateDetails? {
        return null
    }

    @Throws(ICommunicationPartyServiceUpdateServiceDetailsValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceUpdateServiceDetailsGenericFaultFaultFaultMessage::class)
    override fun updateServiceDetails(t: WSServiceUpdate) {
    }

    @Throws(ICommunicationPartyServiceGetCertificateValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceGetCertificateGenericFaultFaultFaultMessage::class)
    override fun getCertificate(ldapUrl: String): ByteArray {
        return ByteArray(0)
    }

    @Throws(ICommunicationPartyServiceGetCertificateForEncryptionValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceGetCertificateForEncryptionGenericFaultFaultFaultMessage::class)
    override fun getCertificateForEncryption(herId: Int): ByteArray {
        return ByteArray(0)
    }

    @Throws(ICommunicationPartyServiceGetLdapSearchProvidersValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceGetLdapSearchProvidersGenericFaultFaultFaultMessage::class)
    override fun getLdapSearchProviders(): WSArrayOfLdapSearchProvider? {
        return null
    }

    override fun updateCoverageAreas(herId: Int, coverageAreas: WSArrayOfCoverageInfo) {}
    @Throws(ICommunicationPartyServiceGetCertificateForValidatingSignatureByOrganizationNumberValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceGetCertificateForValidatingSignatureByOrganizationNumberGenericFaultFaultFaultMessage::class)
    override fun getCertificateForValidatingSignatureByOrganizationNumber(organizationNumber: Int): ByteArray {
        return ByteArray(0)
    }

    @Throws(ICommunicationPartyServiceGetDepartmentDetailsGenericFaultFaultFaultMessage::class)
    override fun getDepartmentDetails(herId: Int): WSDepartment? {
        return null
    }

    @Throws(ICommunicationPartyServiceSearchCertificatesForEncryptionValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceSearchCertificatesForEncryptionGenericFaultFaultFaultMessage::class)
    override fun searchCertificatesForEncryption(sn: String, cn: String, ou: String): WSArrayOfCertificateDetails? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetAssosiatedIKSChildOrganizationsGenericFaultFaultFaultMessage::class)
    override fun getAssosiatedIKSChildOrganizations(herId: Int): WSArrayOfOrganization? {
        return null
    }

    @Throws(ICommunicationPartyServiceCreateOrganizationValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceCreateOrganizationGenericFaultFaultFaultMessage::class)
    override fun createOrganization(org: WSOrganizationCreate): WSOrganization? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetChangeLogGenericFaultFaultFaultMessage::class)
    override fun getChangeLog(herId: Int): WSArrayOfEntityLogEntry? {
        return null
    }

    @Throws(ICommunicationPartyServiceCreateOrganizationPersonValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceCreateOrganizationPersonGenericFaultFaultFaultMessage::class)
    override fun createOrganizationPerson(p: WSOrganizationPersonCreate): WSOrganizationPerson? {
        return null
    }

    @Throws(ICommunicationPartyServiceGetChangedCommunicationPartiesInfoGenericFaultFaultFaultMessage::class)
    override fun getChangedCommunicationPartiesInfo(wsArrayOfint: WSArrayOfint): WSArrayOfCommunicationPartyLastChangedInfo? {
        return null
    }

    @Throws(ICommunicationPartyServiceCreateDepartmentValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceCreateDepartmentGenericFaultFaultFaultMessage::class)
    override fun createDepartment(org: WSDepartmentCreate): WSDepartment? {
        return null
    }

    @Throws(ICommunicationPartyServiceSetCommunicationPartyValidValidationFaultFaultFaultMessage::class, ICommunicationPartyServiceSetCommunicationPartyValidGenericFaultFaultFaultMessage::class)
    override fun setCommunicationPartyValid(herId: Int, period: WSPeriod) {
    }

    override fun getAllHFCommpartiesCsv(): String? {
        return null
    }

    @Throws(ICommunicationPartyServiceRemoveCustomAttributeGenericFaultFaultFaultMessage::class)
    override fun removeCustomAttribute(herId: Int, code: WSCode) {
    }
}
